internal/handlers/user.go

```diff
@@ -409,6 +411,47 @@ func (c User) UpdateReadonlyAccess(repo domain.UserRepository) users.UpdateReado
 	}
 }
 
+func (c User) ChangeEmail(repo domain.UserRepository,
+	changeEmailService domain.ChangeEmailService) users.ChangeEmailHandlerFunc {
+	return func(p users.ChangeEmailParams, access interface{}) middleware.Responder {
+		ctx := p.HTTPRequest.Context()
+		userID, err := authentication.GetUserId(access)
+		if err != nil {
+			c.logger.Error("error while getting authorization during changing email", zap.Error(err))
+			return users.NewChangeEmailUnauthorized().
+				WithPayload(buildStringPayload("Can't get authorization"))
+		}
+
+		if p.EmailPatch == nil {
+			c.logger.Error("email patch is nil", zap.Any("access", access))
+			return users.NewChangeEmailDefault(http.StatusBadRequest).
+				WithPayload(buildStringPayload("Email patch is nil"))
+		}
+
+		requestedUser, err := repo.GetUserByID(ctx, userID)
+		if err != nil {
+			c.logger.Error("getting user for changing email failed", zap.Error(err))
+			return users.NewChangeEmailDefault(http.StatusInternalServerError).
+				WithPayload(buildStringPayload("Can't get user by id"))
+		}
+
+		if requestedUser.IsReadonly {
```

Comment:
I believe this change is unnecessary as the access to the endpoint is controlled by the access manager module and can be specified in 'config.json' file.